### PR TITLE
[GEOS-8356] WMS GetFeatureInfo formats text/plain, GeoJSON don't handle time correctly

### PR DESCRIPTION
**Pull request by fernandor777**
## Description

The following formats: text/html, GeoJSON, text/plain don't have any special treatment for date \ times and simply do a to string regardless of the timezone or time format in WMS GetFeatureInfo.

This is PR fixes GeoJSON, text/plain temporal types formatting handling.

Text/html is managed by Freemarker template engine, but checking how to override it.

Issue:
https://osgeo-org.atlassian.net/browse/GEOS-8356

Requires:
https://github.com/geotools/geotools/pull/2477

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)

The following are required only for core and extension modules, while they are welcomed, but not required, for community modules:
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by fernandor777** [comment]
Checking Freemarker documentation, seems like custom Date/Time format handling is supported from 2.3.24 version but Geoserver master uses 2.3.18.  Should we update version on same issue/PR or made this in its own one?

Source:
https://freemarker.apache.org/docs/pgui_config_custom_formats.html#pgui_config_custom_formats_ex_cust_algo_date

File: src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java [hunk 1]
```diff
@@ -460,6 +461,11 @@ private FeaturesInfo encodeSimpleFeatures(
                                     jsonWriter.writeGeom((Geometry) value);
                                 }
                             }
+                        } else if (Date.class.isAssignableFrom(ad.getType().getBinding())
```
Comment (mbarto) [inline src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java]: This is for WFS GetFeature, not for WMS GetFeatureInfo. Isn't this out of scope?

**Comment by mbarto** [comment]
@fernandor777 I would personally make the text/html fix in a separate PR, since updating a libray could make backporting more problematic

File: src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java (hunk 1 above)
Comment (fernandor777) [reply src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java]: In fact this is a bit confusing at first sight, but doing some debugging this WFS class is used by WMS GetFeatureInfo JSON format output as we can see on stack trace obtained on ```org.geoserver.wms.featureinfo.GetFeatureInfoJSONTest.testDateTimeFormattingEnabled()``` test execution:

```
GeoJSONGetFeatureResponse.encodeSimpleFeatures(GeoJSONBuilder, List<FeatureCollection>, boolean, Operation) line: 467	
GeoJSONGetFeatureResponse.writeFeatures(List<FeatureCollection>, Operation, boolean, GeoJSONBuilder) line: 210	
GeoJSONGetFeatureResponse.write(FeatureCollectionResponse, OutputStream, Operation) line: 133	
GeoJSONGetFeatureResponse(WFSGetFeatureOutputFormat).write(Object, OutputStream, Operation) line: 200	
GeoJSONFeatureInfoResponse.write(FeatureCollectionType, GetFeatureInfoRequest, OutputStream) line: 49	
GetFeatureInfoResponse.write(Object, OutputStream, Operation) line: 99	
Dispatcher.response(Object, Request, Operation) line: 1031	
```

Comment (mbarto) [reply src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONGetFeatureResponse.java]: Fine, I was suspecting that, but needed a confirm :-)

**Comment by nmco** [comment]
@fernandor777 anythign missing here, @mbarto? I restarted the builds.

**Comment by mbarto** [comment]
I am good. +1 for me.

**Pull request merged**